src/screens/NuxLocationScreen.js

```diff
@@ -0,0 +1,192 @@
+  },
+  iconContainer: {
+    alignSelf: 'center',
+  },
+  input: {
+    color: COLORS.textInverse,
+  },
+  label: {
+    color: COLORS.textInverse,
+  },
+  validationText: {
+    color: COLORS.danger2,
+  },
+});
+
+type InjectedProps = {|
+  onContinuePress: (location?: Location) => void | Promise<any>,
+|};
+
+@flatNavigationParamsAndScreenProps
+@observer
+class NuxLocation extends InjectedComponent<InjectedProps> {
+  // We use LocationStore.count() in boundaries  of this component in two places
+  // here and in LocationPicker. LocationPicker flushes the cache of LocationStore
+  // and it leads to infinite loop, so we freeze countLoader for the current case.
+  // To avoid this crutch we need to implement more robust cache control in our
+  // DAO
+  @observable _freezedLocationsCountLoader: LoadObject<number> = null;
+
+  componentDidMount() {
+    when(
+      () => this._locationsCountLoader.hasValue(),
+      () => {
+        this._freezedLocationsCountLoader = this._locationsCountLoader;
+      },
+    );
+  }
+
+  @computed
+  get _locationsCountLoader(): LoadObject<number> {
+    return this._freezedLocationsCountLoader || LocationStore.count();
+  }
+
+  render() {
+    return (
+      <LoaderComponent
+        loadedComponent={LoadedComponent}
+        loader={this._locationsCountLoader}
+        loadingComponent={LoadingComponent}
+        onContinuePress={this.injectedProps.onContinuePress}
+      />
+    );
+  }
+}
+
+const LoadingComponent = () => (
+  <Fragment>
+    <Header showBackButton title="1. Setup location" />
+    <LoadingIndicator color={COLORS.secondary} style={styles.container} />
+  </Fragment>
+);
+
+type LoadedComponentProps = {|
+  onContinuePress: (location?: Location) => void | Promise<any>,
+  value: number,
+|};
+
+@observer
+class LoadedComponent extends React.Component<LoadedComponentProps> {
+  _isLocationLoadingToggleStore: ToggleStore = new ToggleStore();
+
+  @observable _selectedLocation: ?Location = null;
+
+  @action
+  _onLocationSelect = (location: ?Location) => {
+    this._selectedLocation = location;
+  };
+
+  _onContinuePress = async (): Promise<void> => {
+    const { value: locationsCount, onContinuePress } = this.props;
+    switch (locationsCount) {
+      case 0: {
+        onContinuePress();
+        break;
+      }
+      case 1: {
+        this._isLocationLoadingToggleStore.toggleOn();
+        const locationLoaders = await waitForLoaded(() =>
+          LocationStore.getMany(),
+        );
+
+        this._isLocationLoadingToggleStore.toggleOff();
+        onContinuePress(locationLoaders[0].getValueEnforcing());
+        break;
+      }
+      default: {
+        onContinuePress(this._selectedLocation);
+      }
+    }
+  };
+
+  render() {
+    const { value: locationsCount } = this.props;
+
+    const hasNoLocation = locationsCount === 0;
+    const hasOneLocation = locationsCount === 1;
+    const hasManyLocations = locationsCount > 1;
+
+    return (
```

Comment:
Maybe we change this flow so

* zero locations - just show the creation form
* one location - ideally it just takes you to the device setup but if we can't do that... "You've set up a location .." <button>use this</button><button>Create new location</button>
* more locations - same as one location with an additional selector.

Reply:
Hmm.. I guess keep it as is for now. I'll test the flow and let you know if it needs changes.